src/Analyzer.Utilities/Extensions/IMethodSymbolExtensions.cs

```diff
@@ -0,0 +1,120 @@
+                   IsObjectMethodOverride(method);
+        }
+
+        /// <summary>
+        /// Checks if the given method overrides Object.GetHashCode.
+        /// </summary>
+        public static bool IsGetHashCodeOverride(this IMethodSymbol method)
+        {
+            return method.IsOverride &&
+                   method.Name == WellKnownMemberNames.ObjectGetHashCode &&  
+                   method.ReturnType.SpecialType == SpecialType.System_Int32 &&
+                   method.Parameters.Length == 0 &&
+                   IsObjectMethodOverride(method);
+        }
+
+        /// <summary>
+        /// Checks if the given method overrides Object.ToString.
+        /// </summary>
+        public static bool IsToStringOverride(this IMethodSymbol method)
+        {
+            return (method != null &&
+                    method.ReturnType.SpecialType == SpecialType.System_String &&
+                    method.Name == WellKnownMemberNames.ObjectToString &&
+                    method.Parameters.Length == 0 &&
+                    IsObjectMethodOverride(method));
+        }
+
+        /// <summary>
+        /// Checks if the given method overrides a method from System.Object
+        /// </summary>
+        private static bool IsObjectMethodOverride(IMethodSymbol method)
+        {
+            var overriddenMethod = method.OverriddenMethod;
+            while (overriddenMethod != null)
+            {
+                if (overriddenMethod.ContainingType.SpecialType == SpecialType.System_Object)
+                {
+                    return true;
+                }
+
+                overriddenMethod = overriddenMethod.OverriddenMethod;
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Checks if the given method is a Finalizer implementation.
+        /// </summary>
+        public static bool IsFinalizer(this IMethodSymbol method)
+        {
+            if (method.MethodKind == MethodKind.Destructor)
+            {
+                return true; // for C#
+            }
+
+            if (method.Name != "Finalize" || method.Parameters.Length != 0 || !method.ReturnsVoid)
+            {
+                return false;
+            }
+
+            var overridden = method.OverriddenMethod;
+            if (overridden == null)
+            {
+                return false;
+            }
+
+            for (var o = overridden.OverriddenMethod; o != null; o = o.OverriddenMethod)
+            {
+                overridden = o;
+            }
+
+            return overridden.ContainingType.SpecialType == SpecialType.System_Object; // it is object.Finalize
+        }
+
+        /// <summary>
+        /// Checks if the given method implements IDisposable.Dispose()
+        /// </summary>
+        public static bool IsDisposeImplementation(this IMethodSymbol method, Compilation compilation)
+        {
+            if (method.Name != "Dispose")
```

Comment:
True. Changing this would mean that I have to do the more expensive semantic check for every method. I think the case of implementing with a different name is not worth that cost. I'll keep it simple for now and if we hear about issues about different names then we can revisit that.